e2e/jsonschema/match.go

```diff
@@ -66,9 +66,16 @@ func MatchEntities(data []*sdk.EntityData, schemaFileByJobByType map[string]Even
 	}
 
 	for _, entityData := range data {
-		metric := mergeMetricSets(entityData.Metrics)
+		// For easier integration testing, all the different metricSets for each entity are merged into one, which is
+		// then compared with the JsonSchema.
+		// mergeMetricSets errors if this merge would overwrite attributes so the semantics of the tests don't change.
+		metrics, err := mergeMetricSets(entityData.Metrics)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("merging metricSets for %q: %w", entityData.Metadata.Name, err))
+			continue
+		}
```

Comment:
We made some discoveries regarding this, and the comments are not 100% accurate: Having split samples does have an effect. Maybe it's not worth it to address it here though.

Reply:
Ah yeah, with SDKv3 I think this can be removed now IIUC.